js/pages/home/routes.js

```diff
@@ -16,7 +16,8 @@ define(
         '/welcome/:token': new Route((token) => {
           require(['welcome'], function () {
             authApi.token(token);
-            document.location = "#/welcome";
+            // application must be reloaded to call bootstrap function again
+            document.location.replace("/");
```

Comment:
Anyway, this changes affects other OAuth providers such as GitHub or Facebook.

Reply:
in fact other oauth methods also need second attempt or reload to work correctly